src/main/java/com/hubspot/jinjava/lib/filter/LogFilter.java

```diff
@@ -0,0 +1,120 @@
+    })
+public class LogFilter implements Filter {
+
+  private static final MathContext PRECISION = new MathContext(50);
+
+  @Override
+  public Object filter(Object object, JinjavaInterpreter interpreter, String... args) {
+
+    // default to e
+    Double root = null;
+    if (args.length > 0 && args[0] != null) {
+      Double tryRoot = Doubles.tryParse(args[0]);
+      if (tryRoot == null) {
+        throw new InvalidArgumentException(interpreter, this, InvalidReason.NUMBER_FORMAT, 0, args[0]);
+      }
+
+      root = tryRoot;
+    }
+
+    if (object instanceof Integer) {
+      return calculateLog(interpreter, (Integer) object, root);
+    }
+    if (object instanceof Float) {
+      return calculateLog(interpreter, (Float) object, root);
+    }
+    if (object instanceof Long) {
+      return calculateLog(interpreter, (Long) object, root);
+    }
+    if (object instanceof Short) {
+      return calculateLog(interpreter, (Short) object, root);
+    }
+    if (object instanceof Double) {
+      return calculateLog(interpreter, (Double) object, root);
+    }
+    if (object instanceof Byte) {
+      return calculateLog(interpreter, (Byte) object, root);
+    }
+    if (object instanceof BigDecimal) {
+      return calculateBigLog(interpreter, (BigDecimal) object, root);
+    }
+    if (object instanceof BigInteger) {
+      return calculateBigLog(interpreter, new BigDecimal((BigInteger) object), root);
+    }
+    if (object instanceof String) {
+      try {
+        return calculateBigLog(interpreter, new BigDecimal((String) object), root);
+      } catch (NumberFormatException e) {
+        throw new InvalidInputException(interpreter, this, InvalidReason.NUMBER_FORMAT, object.toString());
+      }
+    }
+
+    return object;
+  }
+
+  private double calculateLog(JinjavaInterpreter interpreter, double num, Double base) {
+
+    checkArguments(interpreter, num, base);
+
+    if (base == null) {
+      return Math.log(num);
+    }
+
+    return BigDecimalMath.log(new BigDecimal(num), PRECISION)
+        .divide(BigDecimalMath.log(new BigDecimal(base), PRECISION), RoundingMode.HALF_EVEN)
+        .doubleValue();
+  }
+
+  private BigDecimal calculateBigLog(JinjavaInterpreter interpreter, BigDecimal num, Double base) {
+
+    checkArguments(interpreter, num.doubleValue(), base);
+
+    if (base == null) {
+      return BigDecimalMath.log(num, PRECISION);
+    }
+
+    return BigDecimalMath.log(num, PRECISION)
+        .divide(BigDecimalMath.log(new BigDecimal(base), PRECISION), RoundingMode.HALF_EVEN);
+  }
+
+  private void checkArguments(JinjavaInterpreter interpreter, double num, Double base) {
+
+    if (num <= 0) {
+      throw new InvalidInputException(interpreter, this, InvalidReason.POSITIVE_NUMBER, num);
+    }
+
+    if (base != null && base <= 0) {
+      throw new InvalidArgumentException(interpreter, this, InvalidReason.POSITIVE_NUMBER, 0, base);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return "log";
```

Comment:
I'm not thrilled about the very generic name of this, but it makes sense to be compatible with ansible.